test/05-webresource.test.ts

```diff
@@ -0,0 +1,472 @@
+						headers: {
+							Accept: '*/*',
+						},
+					})
+					.expect(200);
+
+				const receivedSize = photoRes.length;
+				expect(receivedSize).to.equals(newFileSize);
+				const realImage = await fs.readFile(newFilePath);
+				const diff = realImage.compare(photoRes);
+				expect(diff).to.be.eq(0);
+
+				await expectToExist(newFileName);
+				await expectNotToExist(fileKey);
+			});
+
+			it(`should not create an ${resourcePath} with a logo image when content type is not an image`, async () => {
+				await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, filePath, { filename, contentType: 'text/csv' })
+					.expect(400);
+			});
+
+			it(`does not change old ${resourcePath} in storage updating other field that is not webresource`, async () => {
+				const { body: organization } = await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, filePath, { filename, contentType })
+					.expect(201);
+
+				await supertest(testLocalServer)
+					.patch(`/${resourceName}/organization(${organization.id})`)
+					.field('name', 'Peter')
+					.expect(200);
+
+				const getRes = await supertest(testLocalServer)
+					.get(`/${resourceName}/organization(${organization.id})`)
+					.expect(200);
+
+				expect(getRes.body.d[0].name).to.equals('Peter');
+
+				const { body: photoRes } = await supertest(
+					getRes.body.d[0][resourcePath].href,
+				)
+					.get('')
+					.set({
+						responseType: 'arraybuffer',
+						headers: {
+							Accept: '*/*',
+						},
+					})
+					.expect(200);
+
+				const receivedSize = photoRes.length;
+				expect(receivedSize).to.equals(fileSize);
+				const realImage = await fs.readFile(filePath);
+				const diff = realImage.compare(photoRes);
+				expect(diff).to.be.eq(0);
+			});
+
+			it(`deletes ${resourcePath} if transaction size rule is not respected`, async () => {
+				const uniqueFilename = `${randomUUID()}_${filename}`;
+				const { largeStream } = await getLargeFileStream(
+					1024 * 1024 * 600,
+					filePath,
+				);
+				const res = await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, largeStream, {
+						filename: uniqueFilename,
+						contentType,
+					})
+					.expect(400);
+
+				expect(res.body).to.include('It is necessary that');
+				await expectNotToExist(uniqueFilename);
+			});
+
+			it(`deletes ${resourcePath} if content type rule is not respected`, async () => {
+				const uniqueFilename = `${randomUUID()}_${filename}`;
+
+				const res = await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, filePath, {
+						filename: uniqueFilename,
+						contentType: 'text/csv',
+					})
+					.expect(400);
+
+				expect(res.body).to.include('It is necessary that');
```

Comment:
@fisehara My idea here is just to have a way to differentiate the 400 resulted from the rules vs the 400 resulting from a file exceeding the environment variable, do you have any other idea for this?

Reply:
@otaviojacobi This differentiation is fine, my thought is about maintainability / readability. When I run the tests and see that it should return `It is necessary that` but doesn't, where do I search for this. That's why I meant it should have the full rule string to check, so that it's clear where to find the cause for this test.